pinot-spi/src/main/java/org/apache/pinot/spi/utils/builder/TableConfigBuilder.java

```diff
@@ -103,6 +103,8 @@ public class TableConfigBuilder {
   private List<StarTreeIndexConfig> _starTreeIndexConfigs;
   private List<String> _jsonIndexColumns;
   private boolean _aggregateMetrics;
+  private boolean _optimizeDictionaryForMetrics;
```

Comment:
`_noDictionarySizeRatioThreshold` is accepted only when `_optimizeDictionaryForMetrics` is set as `true`? 

Can you clarify the dependency on these configs, if any? Perhaps add javadocs here so the next contributors understand the relationship.

We don't have a good way to document configs atm. javadocs can help in the interim

Reply:
Yes, _noDictionarySizeRatioThreshold is relevant only when _optimizeDictionaryForMetrics is true. Will add java docs.